mne_bids/write.py

```diff
@@ -879,6 +883,14 @@ def write_raw_bids(raw, bids_basename, output_path, events_data=None,
     acquisition, task, run = params['acq'], params['task'], params['run']
     kind = _handle_kind(raw)
 
+    if anonymize:
+        if ext != '.fif':
+            warn('Converting to FIF to anonymize')
+            ext = '.fif'
+        raw.info = anonymize_info(raw.info)
+        raw_fname = 'anonymize_tmp-raw.fif'
```

Comment:
can we avoid this intermediate writing to disk? wouldn't this slow things down if I had many big files?

Reply:
Yes, I think this would slow things down but unless write_raw_bids would save instead of write every time I'm not sure if I know a better way...

Since it's only one file and it gets removed right after on the bright side there is less likely to be disc space or RAM errors.

Reply:
@jasmainak in regards to this, since the conversion tool is made, one option would be to do exactly that, no longer take them from disc but instead just save them out to BV or FIF. I think the main argument against that is that people with EDF etc. may not want to convert but it would save headaches as far as saving the original data, saving an anonymized version and saving a BIDS version and instead would just convert directly. It's rather a big decision about whether the culture of mne-bids supports all BIDS-supported files equally or makes FIF/BV the mne-bids-supported formats.  cc @agramfort

Reply:
If we keep the original data format, we don't guarantee anonymization. Let's make sure it's clear in the docstrings that conversion will happen if it's not fif or BV.

Reply:
Ok, I think that part is pretty clear with the warnings but I can clarify more in the documentation.